packages/cli/src/commands/experimental/setupRealtimeHandler.js

```diff
@@ -0,0 +1,291 @@
+import fs from 'fs'
+import path from 'path'
+
+import { Listr } from 'listr2'
+
+import { generate as generateTypes } from '@redwoodjs/internal/dist/generate/generate'
+import { getConfigPath } from '@redwoodjs/project-config'
+import { errorTelemetry } from '@redwoodjs/telemetry'
+
+import { getPaths, transformTSToJS, writeFile } from '../../lib'
+import c from '../../lib/colors'
+import { isTypeScriptProject } from '../../lib/project'
+
+import { command, description, EXPERIMENTAL_TOPIC_ID } from './setupRealtime'
+import { printTaskEpilogue, isServerFileSetup } from './util'
+
+export async function handler({ force, verbose }) {
+  const redwoodPaths = getPaths()
+  const ts = isTypeScriptProject()
+
+  const realtimeLibFilePath = path.join(
+    redwoodPaths.api.lib,
+    `realtime.${isTypeScriptProject() ? 'ts' : 'js'}`
+  )
+
+  const tasks = new Listr(
+    [
+      {
+        title: 'Confirmation',
+        task: async (_ctx, task) => {
+          const confirmation = await task.prompt({
+            type: 'Confirm',
+            message:
+              'Realtime is currently an experimental RedwoodJS feature. Continue?',
+          })
+
+          if (!confirmation) {
+            throw new Error('User aborted')
+          }
+        },
+      },
+      {
+        title: 'Checking for realtime environment prerequisites ...',
+        task: () => {
+          isServerFileSetup()
+        },
+      },
+      {
+        title: 'Adding the realtime api lib ...',
+        task: () => {
+          const serverFileTemplateContent = fs.readFileSync(
+            path.resolve(__dirname, 'templates', 'realtime.ts.template'),
+            'utf-8'
+          )
+
+          const setupScriptContent = ts
+            ? serverFileTemplateContent
+            : transformTSToJS(realtimeLibFilePath, serverFileTemplateContent)
+
+          return [
+            writeFile(realtimeLibFilePath, setupScriptContent, {
+              overwriteExisting: force,
+            }),
+          ]
+        },
+      },
+      {
+        title: 'Adding Countdown example subscription ...',
```

Comment:
Should we have some sort of prompt or option to avoid generating the examples? I'm not that fussed about it just in case someone doesn't want the new sdls, services and subscriptions.

Reply:
Good idea. Am implementing.